pkg/storage/etcd/etcd_helper.go

```diff
@@ -558,6 +576,15 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 			ttl = *newTTL
 		}
 
+		// Since update object may have a resourceVersion set, we need to clear it here.
+		if h.versioner != nil {
+			if err := h.versioner.UpdateObject(ret, meta.Expiration, 0); err != nil {
```

Comment:
@smarterclayton - sorry, I'm afraid I don't understand.
You suggest leaving as is or changing (to not use versioner and just work on meta.Accessor)?